### PR TITLE
extract username generation to an overridable method

### PR DESCRIPTION
**Pull request by anlutro**
Currently you can override `username_algo` but that assumes that you want to generate the username based on the email. This is not always the case.

Adding a method `get_username` lets us generate usernames based on `sub`, for example.

### CODE REVIEW

**Comment by anlutro** [comment]
Tests seem to fail on Travis because `wheel` stopped supporting Python 3.3?

File: mozilla_django_oidc/auth.py [hunk 1]
```diff
@@ -77,22 +77,25 @@ def verify_claims(self, claims):
 
     def create_user(self, claims):
         """Return object for a newly created user account."""
-        # bluntly stolen from django-browserid
```
Comment (johngian) [inline mozilla_django_oidc/auth.py]: Let's also keep the reference in ``get_username``.

File: mozilla_django_oidc/auth.py [hunk 2]
```diff
@@ -77,22 +77,25 @@ def verify_claims(self, claims):
...
         email = claims.get('email')
         if not email:
             return None
 
+        username = self.get_username(claims)
+
+        return self.UserModel.objects.create_user(username, email)
+
+    def get_username(self, claims):
+        """Get the local username based on claims."""
```
Comment (johngian) [inline mozilla_django_oidc/auth.py]: I am not sure if ``local username`` makes a lot of sense. I would prefer something like:

```Generate username based on claims.```

**Comment by johngian** [comment]
Thanks @anlutro for the PR. Code looks really straightforward and it doesn't cause any incompatibilities. I added a couple of comments regarding comments/docstrings.

**Comment by anlutro** [comment]
Feedback incorporated

**Comment by johngian** [comment]
Closing this one after #242

**Pull request closed without merging**